appservice/src/deploy/waitForDeploymentToComplete.ts

```diff
@@ -8,24 +8,31 @@ import { DeployResult, LogEntry } from 'vscode-azurekudu/lib/models';
 import { ext } from '../extensionVariables';
 import { localize } from '../localize';
 import { SiteClient } from '../SiteClient';
+import { delay } from '../utils/delay';
 import { nonNullProp } from '../utils/nonNull';
 import { formatDeployLog } from './formatDeployLog';
 
-export async function waitForDeploymentToComplete(client: SiteClient, kuduClient: KuduClient, pollingInterval: number = 5000): Promise<void> {
+export async function waitForDeploymentToComplete(client: SiteClient, kuduClient: KuduClient, expectedId?: string, pollingInterval: number = 5000): Promise<void> {
     const alreadyDisplayedLogs: string[] = [];
     let nextTimeToDisplayWaitingLog: number = Date.now();
-    let permanentId: string | undefined;
     let initialStartTime: Date | undefined;
     let deployment: DeployResult | undefined;
+    let permanentId: string | undefined;
+    // a 30 second timeout period to let Kudu initialize the deployment
+    const maxTimeToWaitForExpectedId: number = Date.now() + 30 * 1000;
 
     // tslint:disable-next-line:no-constant-condition
     while (true) {
-        [deployment, permanentId, initialStartTime] = await getLatestDeployment(kuduClient, permanentId, initialStartTime);
+        [deployment, permanentId, initialStartTime] = await tryGetLatestDeployment(kuduClient, permanentId, initialStartTime, expectedId);
+        if ((deployment === undefined || !deployment.id)) {
+            if (expectedId && Date.now() < maxTimeToWaitForExpectedId) {
+                ext.outputChannel.appendLine(formatDeployLog(client, localize('waitingForBuild', 'Waiting for build to trigger...')));
```

Comment:
On the first deploy, there is a considerable amount of time where nothing is displayed in the output channel, so I thought this might be useful.  Thoughts?

Reply:
A message is fine, but I would prefer if it was consistent with zipDeploy:
https://github.com/Microsoft/vscode-azuretools/blob/master/appservice/src/deploy/deployZip.ts#L36